governance/xc_admin/packages/xc_admin_frontend/components/tabs/Proposals/utils.ts

```diff
@@ -0,0 +1,100 @@
+import { PythCluster } from '@pythnetwork/client'
+import { AccountMeta } from '@solana/web3.js'
+import { MultisigAccount, TransactionAccount } from '@sqds/mesh/lib/types'
+import {
+  ExecutePostedVaa,
+  MultisigInstruction,
+  MultisigParser,
+  PythGovernanceActionImpl,
+  SetDataSources,
+  WormholeMultisigInstruction,
+} from 'xc_admin_common'
+
+export type ProposalStatus =
+  | 'active'
+  | 'executed'
+  | 'cancelled'
+  | 'rejected'
+  | 'expired'
+  | 'executeReady'
+  | 'draft'
+  | 'unkwown'
+
+export const getProposalStatus = (
+  proposal: TransactionAccount | undefined,
+  multisig: MultisigAccount | undefined
+): ProposalStatus => {
+  if (multisig && proposal) {
+    const onChainStatus = Object.keys(proposal.status)[0]
+    return proposal.transactionIndex <= multisig.msChangeIndex &&
+      (onChainStatus == 'active' || onChainStatus == 'draft')
+      ? 'expired'
+      : (onChainStatus as ProposalStatus)
+  } else {
+    return 'unkwown'
+  }
+}
+
+/**
+ * Sorts the properties of an object by their values in ascending order.
+ *
+ * @param {Record<string, number>} obj - The object to sort. All property values should be numbers.
+ * @returns {Record<string, number>} A new object with the same properties as the input, but ordered such that the property with the largest numerical value comes first.
+ *
+ * @example
+ * const obj = { a: 2, b: 3, c: 1 };
+ * const sortedObj = sortObjectByValues(obj);
+ * console.log(sortedObj); // Outputs: { b: 3, a: 2, c: 1 }
+ */
+const sortObjectByValues = (obj: Record<string, number>) => {
+  const sortedEntries = Object.entries(obj).sort(([, a], [, b]) => b - a)
+  const sortedObj: Record<string, number> = {}
+  for (const [key, value] of sortedEntries) {
+    sortedObj[key] = value
+  }
+  return sortedObj
```

Comment:
how confident are you that the of insertion is the order of retrieval? Quick google says that maybe`Map` is better for this

Reply:
100%. For string keys it's always the same order. Also this is just for the sake of better UI, it really doesn't matter if the order is nor correct.